{"uri":"file:///Users/vega/Code/huli/manager-web/src/vendor/phpunit/phpunit/src/Framework/Constraint/Not.php","root":{"kind":0,"name":"","children":[{"kind":1,"name":"PHPUnit_Framework_Constraint_Not","location":[13,0,155,1],"associated":[{"kind":1,"name":"PHPUnit_Framework_Constraint"}],"description":"Logical NOT.","children":[{"kind":16,"name":"$constraint","modifiers":2,"location":[18,14,18,25],"scope":"PHPUnit_Framework_Constraint_Not","type":"PHPUnit_Framework_Constraint"},{"kind":32,"name":"__construct","modifiers":1,"location":[23,4,32,5],"scope":"PHPUnit_Framework_Constraint_Not","children":[{"kind":128,"name":"$constraint","location":[23,32,23,43],"scope":"__construct","type":"PHPUnit_Framework_Constraint"}]},{"kind":32,"name":"negate","modifiers":33,"location":[39,4,68,5],"scope":"PHPUnit_Framework_Constraint_Not","type":"string","children":[{"kind":128,"name":"$string","location":[39,34,39,41],"scope":"negate","type":"string"}]},{"kind":32,"name":"evaluate","modifiers":1,"location":[88,4,99,5],"description":"Evaluates the constraint for parameter $other\n\nIf $returnResult is set to false (the default), an exception is thrown\nin case of a failure. null is returned otherwise.\n\nIf $returnResult is true, the result of the evaluation is returned as\na boolean value instead: true in case of success, false in case of a\nfailure.","scope":"PHPUnit_Framework_Constraint_Not","type":"mixed","children":[{"kind":128,"name":"$other","location":[88,29,88,35],"description":"Value or object to evaluate.","scope":"evaluate","type":"mixed"},{"kind":128,"name":"$description","location":[88,37,88,54],"description":"Additional information about the test","scope":"evaluate","value":"''","type":"string"},{"kind":128,"name":"$returnResult","location":[88,56,88,77],"description":"Whether to return a result or throw an exception\n","scope":"evaluate","value":"false","type":"bool"}]},{"kind":32,"name":"failureDescription","modifiers":2,"location":[111,4,124,5],"description":"Returns the description of the failure\n\nThe beginning of failure messages is \"Failed asserting that\" in most\ncases. This method should return the second part of that sentence.","scope":"PHPUnit_Framework_Constraint_Not","type":"string","children":[{"kind":128,"name":"$other","location":[111,42,111,48],"description":"Evaluated value or object.\n","scope":"failureDescription","type":"mixed"}]},{"kind":32,"name":"toString","modifiers":1,"location":[131,4,144,5],"description":"Returns a string representation of the constraint.","scope":"PHPUnit_Framework_Constraint_Not","type":"string"},{"kind":32,"name":"count","modifiers":1,"location":[151,4,154,5],"description":"Counts the number of constraint elements.","scope":"PHPUnit_Framework_Constraint_Not","type":"int"}]}]}}